pkg/operator/ceph/object/spec.go

```diff
@@ -444,11 +444,15 @@ func (c *clusterConfig) CheckRGWKMS() (bool, error) {
 			return false, err
 		}
 		secretEngine := c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]
-		kvVers := c.store.Spec.Security.KeyManagementService.ConnectionDetails[vault.VaultBackendKey]
 
 		// currently RGW supports kv(version 2) and transit secret engines in vault
```

Comment:
My idea was to skip `VaultBackendKey` entirely for RGW since it only supports `v2` and this function checks for secretengine kv or transit(similar to older version of this API). Even if it is defined we ignore that and have documented that info. If the user enables `kv-v1` at the vault backend and tries to use that path, then it is a misconfiguration from his end

Reply:
But shouldn't we return an error if there is a misconfiguration?

Reply:
if it is not required to add the
```
path "sys/mounts" {
     capabilities = ["read"]
}
```
to the policy for vault token, then I am fine this change in for CheckRGWKMS()
Currently, RGW does not need that permission I am not sure whether using `vault client` from rook may impose that permission or not

Reply:
I think it does since `"sys/mounts"` is where we will auto-detect the KV backend version.

Reply:
That's I am not quite sure whether it is right to give the above permission to RGW. RGW just need permission to read keys from the vault server in a specified path nothing else, vault server management is done by the user. 
In upstream RGW is moving away from that dependency as well with help of vault agent [AFAIK](https://docs.ceph.com/en/latest/radosgw/vault/)
So adding this permission for RGW to check the unsupported version of the kv engine does not make much sense for me. Since we have separate CRs for OSD and RGW encryption, it is okay to assume for kv secret engine RGW expects to have v2 and if does not it is misconfiguration from user.

Reply:
Ok point taken, I'll remove the path check and we just fail if the backend is not v2. No auto-detection needed.

Reply:
Actually one more thing, we need to be able to handle cases where `VAULT_BACKEND` is not present in `store.Spec.Security.KeyManagementService.ConnectionDetails`. This is what this PR is about, so without auto-detection, how de you suggest we handle this?

Reply:
IMO we can ignore that the value, remove the version-check in `CheckRGWKMS()`, even if user defines the value just ignore it and assume it is always `kv-v2`. If user enables `kv-v1` and gives  `VAULT_SECRET_ENGINE` as kv(misconfiguration situation) RGW will come up but when he tries to encrypt the object it will fail(most probably key not found error).

Reply:
Ok, I'll assume kv-v2 and log a warning then if `VAULT_BACKEND` is not specified in the `store.Spec.Security.KeyManagementService.ConnectionDetails`.